### PR TITLE
chore: Run conformance tests during CI

### PR DESCRIPTION
**Pull request by mtraver**
Currently HTTP and CloudEvents without mapping will run. Next task is to update event mapping and enable that in the conformance test config.

### CODE REVIEW

File: .github/workflows/conformance.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Ruby Conformance CI
+on: push
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        ruby-version: [2.5, 2.6, 2.7]
```
Comment (dazuma) [inline .github/workflows/conformance.yml:8]: Can we include 3.0?

File: .github/workflows/conformance.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Ruby Conformance CI
+on: push
```
Comment (dazuma) [inline .github/workflows/conformance.yml:2]: Should we use the same triggers as the unit and lint tests? (i.e. run on pull requests and merges to main). I'd also like to include workflow_dispatch so admins can trigger manually.

```
on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
  workflow_dispatch:
```

File: .github/workflows/conformance.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      with:
+        go-version: '1.15'
+    - name: Setup Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby-version }}
+    - name: Bundle install
+      run: 'bundle install'
+    - name: Run HTTP conformance tests
+      uses: GoogleCloudPlatform/functions-framework-conformance/action@v0.3.6
```
Comment (dazuma) [inline .github/workflows/conformance.yml:23]: Is it expected that we need to call out the specific version? (i.e. is this something we'll have to keep updated?)

File: .github/workflows/conformance.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: Ruby Conformance CI
+on: push
+jobs:
+  build:
+    runs-on: ubuntu-18.04
```
Comment (dazuma) [inline .github/workflows/conformance.yml:5]: Does this need to be 18.04 or can we use latest?

File: .github/workflows/conformance.yml (hunk 1 above)
Comment (mtraver) [reply .github/workflows/conformance.yml:8]: Done

File: .github/workflows/conformance.yml (hunk 2 above)
Comment (mtraver) [reply .github/workflows/conformance.yml:2]: Done

File: .github/workflows/conformance.yml (hunk 3 above)
Comment (mtraver) [reply .github/workflows/conformance.yml:23]: As far as I can tell you have to specify some version, so I'll change it to v0.3.

File: .github/workflows/conformance.yml (hunk 4 above)
Comment (mtraver) [reply .github/workflows/conformance.yml:5]: Latest should be fine.

File: .github/workflows/conformance.yml (hunk 3 above)
Comment (mtraver) [reply .github/workflows/conformance.yml]: Turns out that doesn't work. It'll have to be pinned for now.

**Review by dazuma — APPROVED** [review]
Thanks much!!

**Pull request merged**